Source/CheatSheet/cheatsheet.md

```diff
@@ -62,79 +63,106 @@ The retrieve expression is the central construct for accessing clinical informat
 
-```bnf
-<primary-source> <alias>
-  <with-or-without-clauses>
-  <where-clause>
-  <return-clause>
-  <sort-clause>
+A function in CQL is a named expression that is allowed to take any number of arguments. A function can be invoked directly by name. A colon must end the quoted identifier.
+
+```cql
+define function MostRecent(observations List<Observation>):
+  Last(
+    observations O
+      sort by issued
+  )
 ```
 
-TODO: Should probably organize discussions about each clause here
+In this example, the function takes a list of Observations, sorts them by their issued date and returns the last one that has been issued.
 
-TODO: Identifiers should probably be introduced before queries?
-TODO: The link is to "identifier", but if this is about "identifiers" vs "strings", we should probably have links to both?
-TODO: Probably call it "Identifiers vs Strings", rather than "Quotation Syntax"?
-### [**Quotation Syntax**](https://cql.hl7.org/19-l-cqlsyntaxdiagrams.html#identifier) 
+## Strings vs Identifiers
 
-Strings have single quotes (including string representation of code values)
+**Strings use single quotation marks.**
 
-``` cql
-'John Doe'
-'g/dl'
-'male'
+[Single Quotes](https://cql.hl7.org/19-l-cqlsyntaxdiagrams.html#string) 
+
+Single quotes are limited to string literals
+1) text
+```cql
+define "string literal": 'hello'
 ```
 
-TODO: Probably need to introduce "Identifiers" somewhere?
-Identifiers that include spaces or other non-alphnumeric characters have double quotes 
+2) code system and value set object identifiers (OID)
+```cql
+valueset "example":  'urn:oid:2.16.840.1.113883.3.560.100.2'
+```
 
+3) version identifiers
 ```cql
-"Marital Status - Married" // A Concept declaration
-"SNOMED CT" // A CodeSystem declaration
-"Inpatient Encounters" // A defined expression
+valueset "example":  'urn:oid:2.16.840.1.113883.3.560.100.2' version '123'
 ```
 
-TODO: The link is to "identifier", which isn't relevant to any of these topics?
-### [**Bracket Syntax**](https://cql.hl7.org/19-l-cqlsyntaxdiagrams.html#identifier)
+4) code declarations
+```cql
+code "code": '123' from "codesystem identifier"
+```
+
+5) units
+```cql
+define "quantity": 12.0 'L'
+```
+
+**Identifiers may use double quotation marks.**
+
+[Quoted Identifiers](https://cql.hl7.org/19-l-cqlsyntaxdiagrams.html#QUOTEDIDENTIFIER)
+This applies to definitions, functions, valuesets, codes, etc... 
+
+```cql
+"SNOMED CT" // A code system declaration
+"Inpatient Encounters" // A Defined Code
+```
+
+### [**Identifiers**](https://cql.hl7.org/03-developersguide.html#identifiers)
+Identifiers are used to name various elements within the language. There are three types of identifiers in CQL, simple, delimited, and quoted.
+
+```cql
+Foo1 // simple
+`Encounter, Performed` // delimited
+"Inpatient Encounters" // quoted
+```
+
+
+## Bracket Syntax
 
-1. Intervals use [] and ()
+1. [Intervals](https://cql.hl7.org/19-l-cqlsyntaxdiagrams.html#intervalSelector) use [] and ()
```

Comment:
Suggest we add narrative to describe the difference between square bracket and parentheses:

```suggestion
1. [Intervals](https://cql.hl7.org/19-l-cqlsyntaxdiagrams.html#intervalSelector) use [] and (). Following standard mathematics notation, inclusive (closed) boundaries are indicated with square brackets, and exclusive (open) boundaries are indicated with parentheses.
```